### PR TITLE
Update docs with 'Implementation Status' section

### PR DESCRIPTION
**Pull request by dcolligan**
Issue #432

### CODE REVIEW

File: docs/source/status.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.. _status:
+
+---------------------
+Implementation Status
+---------------------
+
+The GA4GH server is currently under active development, and not all features
```
Comment (jeromekelleher) [inline docs/source/status.rst]: "not all features may be fully functional yet" reads a bit funny. Perhaps "several features are not yet fully functional"?

**Comment by jeromekelleher** [comment]
+1 modulo suggested rephrase.

**Comment by dcolligan** [comment]
@jeromekelleher pushed commit to rephrase

**Comment by jeromekelleher** [comment]
Great, thanks @dcolligan. +1

**Comment by dcolligan** [comment]
merge me!

**Pull request merged**